npc_simulator/src/node.cpp

```diff
@@ -26,63 +30,92 @@ typedef bg::model::d2::point_xy<double> Point;
 typedef bg::model::linestring<Point> Line;
 typedef bg::model::polygon<Point> Polygon;
 
-NPCSimulatorNode::NPCSimulatorNode() : nh_(), pnh_("~"), tf_listener_(tf_buffer_)
+NPCSimulatorNode::NPCSimulatorNode()
+: rclcpp::Node("npc_simulator"),
+  tf_buffer_(get_clock()),
+  tf_listener_(tf_buffer_),
+  vehicle_info_(vehicle_info_util::VehicleInfo::create(*this))
 {
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+
   // get parameter
-  pnh_.param<bool>("initial_engage_state", engage_state_, true);
+  engage_state_ = declare_parameter<bool>("initial_engage_state", true);
 
   // get vehicle parameter
-  vehicle_width_ = waitForParam<double>(pnh_, "/vehicle_info/vehicle_width");
-  vehicle_length_ = waitForParam<double>(pnh_, "/vehicle_info/vehicle_length");
-  vehicle_rear_overhang_ = waitForParam<double>(pnh_, "/vehicle_info/rear_overhang");
-  vehicle_base2center_ = vehicle_length_ / 2.0 - vehicle_rear_overhang_;
-
-  dummy_perception_object_pub_ =
-    pnh_.advertise<dummy_perception_publisher::Object>("output/dynamic_object_info", 10, true);
-  debug_object_pub_ = pnh_.advertise<autoware_perception_msgs::DynamicObjectArray>(
-    "output/debug_object_info", 10, true);
+  double wheel_tread{}, left_overhang{}, right_overhang{};
+  assert(get_parameter("wheel_tread", wheel_tread));
+  assert(get_parameter("left_overhang", left_overhang));
+  assert(get_parameter("right_overhang", right_overhang));
+  vehicle_width_ = wheel_tread + left_overhang + right_overhang;
+
+  double wheel_base{}, front_overhang{}, rear_overhang{};
+  assert(get_parameter("wheel_base", wheel_base));
+  assert(get_parameter("front_overhang", front_overhang));
+  assert(get_parameter("rear_overhang", rear_overhang));
+  vehicle_length_ = wheel_base + front_overhang + rear_overhang;
```

Comment:
Where does this change come from? In the ROS1 version `vehicle_length` seems to be an independent parameter.

Reply:
@cvasfi Fixed now when I use vehicle_info properly as @nnmm showed me